### PR TITLE
ci: add GitHub Actions

### PR DESCRIPTION
**Pull request by paulswartz**

### CODE REVIEW

**Comment by paulswartz** [comment]
Successful action: https://github.com/paulswartz/logger_splunk_backend/actions/runs/875814899

File: mix.exs [hunk 1]
```diff
@@ -5,13 +5,17 @@ defmodule LoggerSplunkBackend.Mixfile do
     [
       app: :logger_splunk_backend,
       version: "2.0.0",
```
Comment (losvedir) [inline mix.exs]: Should we bump the version here? And I think the change in Elixir version is a breaking change, so we might even have to go to 3.0.0?

**Review by losvedir — APPROVED** [review]
Just one question about the version, otherwise 👍 .

File: mix.exs (hunk 1 above)
Comment (paulswartz) [reply mix.exs]: Given that this is only changing to support our new CI provider, I wasn't planning to do a release. There's no new functionality here.

Comment (paulswartz) [reply mix.exs]: But that's a good point about the new version: I'll update the CHANGELOG and commit message so we don't forget if we do a release in the future.

**Pull request merged**